src/sdks/core/src/cpp/sdk/cpptest/unit/deviceTest.cpp

```diff
@@ -0,0 +1,286 @@
+    Firebolt::Types::BooleanMap actualValue = Firebolt::IFireboltAccessor::Instance().DeviceInterface().hdcp(&error);
+
+    EXPECT_EQ(error, Firebolt::Error::None);
+    EXPECT_EQ(actualValue, expectedValues);
+}
+
+
+TEST_F(DeviceTest, TestDevicehdr)
+{
+    Firebolt::Error error = Firebolt::Error::None;
+    
+    // Parsing expected JSON values into a BooleanMap
+    nlohmann::json expectedJson = nlohmann::json::parse(jsonEngine->get_value("Device.hdr"));
+    Firebolt::Types::BooleanMap expectedValues;
+
+    for (auto it = expectedJson.begin(); it != expectedJson.end(); ++it) {
+        expectedValues[it.key()] = it.value().get<bool>();
+    }
+     
+    // Getting the actual value from the DeviceInterface
+    Firebolt::Types::BooleanMap actualValue = Firebolt::IFireboltAccessor::Instance().DeviceInterface().hdr(&error);
+
+    EXPECT_EQ(error, Firebolt::Error::None);
+    EXPECT_EQ(actualValue, expectedValues);
+}
+
+
+std::string AudioProfileToString(Firebolt::Types::AudioProfile profile) {
+    switch (profile) {
+        case Firebolt::Types::AudioProfile::STEREO:
+            return "stereo";
+        case Firebolt::Types::AudioProfile::DOLBY_DIGITAL_5_1:
+            return "dolbyDigital5.1+";
+		case Firebolt::Types::AudioProfile::DOLBY_ATMOS:
+            return "dolbyAtmos";
+		default:
+            return "dolbyDigital5.1";
+        
+    }
+} 
+
+TEST_F(DeviceTest, TestDeviceaudio)
+{
+    Firebolt::Error error = Firebolt::Error::None;
+    
+    // Parse expected JSON values
+    nlohmann::json expectedJson = nlohmann::json::parse(jsonEngine->get_value("Device.audio"));
+     
+    // Getting the actual value from the DeviceInterface
+    const Firebolt::Device::AudioProfiles actualValue = Firebolt::IFireboltAccessor::Instance().DeviceInterface().audio(&error);
+
+    // Convert actualValue (AudioProfiles map) to JSON
+    nlohmann::json actualJson;
+    for (const auto& item : actualValue) {
+        std::string profileKey = AudioProfileToString(item.first); // Replace this with the correct conversion function
+        actualJson[profileKey] = item.second;
+    }
+
+    EXPECT_EQ(error, Firebolt::Error::None);
+    EXPECT_EQ(actualJson, expectedJson);
+}
+
+TEST_F(DeviceTest, TestDevicenetwork)
+{
+    Firebolt::Error error = Firebolt::Error::None;
+    
+    // Hardcoded expected values
+    Firebolt::Device::NetworkInfo expectedValues;
+    expectedValues.state = Firebolt::Device::NetworkState::CONNECTED; 
+    expectedValues.type = Firebolt::Device::NetworkType::WIFI;
+     
+    // Getting the actual value from the DeviceInterface
+    Firebolt::Device::NetworkInfo actualValue = Firebolt::IFireboltAccessor::Instance().DeviceInterface().network(&error);
+  
+    // Perform the assertions
+    EXPECT_EQ(error, Firebolt::Error::None);
+    EXPECT_EQ(actualValue.state, expectedValues.state);
+    EXPECT_EQ(actualValue.type, expectedValues.type);
+}
+
+TEST_F(DeviceTest, DeviceScreenResolution)
```

Comment:
Is there a difference between this test and the one at line 243?

Reply:
good spot! there was a correction in test name, the correct name for the other test is `TEST_F(DeviceTest, VideoResolution)`. Now corrected.